### DISCUSSION TITLE
Visual Studio: cannot disable inline suggestions

### DISCUSSION THREAD

**Discussion by drewboardman** [community/community#84995] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

It seems as though there is no way to disable the automatic inline suggestions in Visual Studio. The leads to very annoying user experience. I'd like to disable to suggestion unless I specifically trigger it with the keybinding for inline suggestions.

**Comment by ayim** [comment]
Hey! It's not published but I managed to get this to work in VS:
1) Click the purple lightbulb icon, and uncheck Show Completions for lines of Code
This feature makes the requests to the GitHub Copilot / IntelliCode models as you type
![image](https://github.com/community/community/assets/10951869/661dd92c-5ce5-471f-83ff-3e0901c5dd54)
2) As you want a prediction, you can type Ctrl + Alt + \  or whatever your keybinding is and get a prediction if available.
Generally, if the icon next to "Ready" stops moving and you don't get grey text, that means there's no prediction available at your location.
![image](https://github.com/community/community/assets/10951869/2416b0b3-f145-4391-9c38-2a8ad16624e7)

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio
aayim@microsoft.com

**Reply by drewboardman** [reply]
nice!

**Reply by drewboardman** [reply]
EDIT: upon further inspection - this seems to either be:

 * broken with the latest update
 * only works until you restart the editor.

I currently have all inline options unchecked, and copilot still autosuggests in-line.

![image](https://github.com/community/community/assets/8429384/5801072a-9f5a-4e5b-9332-6170effe1271)
![image](https://github.com/community/community/assets/8429384/c73ea6ab-d4f0-4dc5-bc26-cfc43ae50932)
![image](https://github.com/community/community/assets/8429384/279011cf-3837-4eda-9ee0-cd8b674289e9)

**Reply by ayim** [reply]
Does the behaviour change if you use the option in Tools > Options vs. the purple lightbulb in the bottom left of the editor?
AFAIK the setting in Tools > Options requires restart to take effect, but the one in the purple lightbulb doesn't.
Both settings should be sticky.

Another option you have is to check "Sow completions for lines of code" but also select "Wait for pauses in typing before showing completions"

**Reply by tylerlaprade** [reply]
@drewboardman , this is off-topic, but what font is that? It looks very nice.
